vsg/__main__.py

```diff
@@ -393,13 +409,21 @@ def main():
 
     dStyle = read_predefined_style(commandLineArguments.style)
 
-    configuration = read_configuration_files(dStyle, commandLineArguments)
+    fix_only = settings.open_configuration_file(commandLineArguments.fix_only)
+
+    if commandLineArguments.configuration:
+        configuration = read_configuration_files(dStyle, commandLineArguments)
+
+        dIndent = read_indent_configuration(configuration)
 
-    dIndent = read_indent_configuration(configuration)
+        update_command_line_arguments(commandLineArguments, configuration)
+
+    else:
+        update_command_line_arguments2(commandLineArguments)
 
-    fix_only = open_fix_file(commandLineArguments.fix_only)
```

Comment:
I don't really understand what is happening here, why is `open_fix_file` slightly different from`open_configuration_file`? I have merged them together but might have missed some corner cases.

Reply:
The difference is subtle, essentially just what is reported as an error and the `TypeError` exception.  It will be find to merge them.